### DISCUSSION TITLE
How can I interact with a Button in Godot?

### DISCUSSION THREAD

**Discussion by Elisicc** [community/community#142574] [Programming Help]
### Body

I have an character model and I would really like this to interct with 3D objects like a Button. Does someone have a Idea how I can do that?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Tristan-BS** [comment]
Hi again xDDDD

I did that too a long time ago :0

I just created a scene with paren "StaticBody3D" and gave it a base and Button Object and a Collisionshape3D to interact.
I also created a Script for this scene wich you can see here:
`extends StaticBody3D

var door_open:bool = false
var can_press:bool = true

signal interacted(body)

@export var prompt_message = "Interact"
@export var prompt_action = "interact"
@onready var interacted_message = $"../Player/Interacted Message"
@onready var anim = $"../House/anim"

func _ready():
	interacted_message.text = ""

func interact(body):
	emit_signal("interacted", body)
	#interacted_message.text = "User Interacted"
	
	var door = Help.GetDoorState()
	
	if !door and can_press:
		anim.play("Open Door")
		interacted_message.text = "Door Opened"
		door_open = true
		Help.SetDoorState(door_open)

		can_press = false
		await get_tree().create_timer(1.5).timeout
		can_press = true

	elif door and can_press:
		anim.play("Close Door")
		interacted_message.text = "Door Closed"
		door_open = false
		Help.SetDoorState(door_open)

		can_press = false
		await get_tree().create_timer(1.5).timeout
		can_press = true
	
	await get_tree().create_timer(0.5).timeout
	interacted_message.text = ""
`

This StaticBody is also a Group of "Interactable". 


For the "arm" of the Player i created an Node into the player and a SpringArm3D
Due to the new Godot Version the code of my player changed:
`@export var first_person: bool = false : 
	set(p_value):
		first_person = p_value
		if first_person:
			$CameraManager/Arm.spring_length = 0.0
		else:
			$CameraManager/Arm.spring_length = 6.0`


Hope this helps ( again ). If you need further Informations or a closer look to my Project don't hesitate to ask me :)

**Comment by Elisicc** [comment]
Ohh ok, that helps, thank you very much, I will try it now

**Comment by ebndev** [comment]
Hi @Elisicc & @Tristan-BS, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. 

Thanks for understanding and we hope to see you participate in other discussions in the future!
